### PR TITLE
Use Kafka Confluent lib; pull kafka TLS/auth config from Clowder

### PR DESCRIPTION
**Pull request by kylape**
There are a couple of other commits in this PR:

1. Move source under pkg dir and rework the Dockerfile to make it build faster and with a cleaner image.
2. Add a newline to all JSON log lines to help with reading stdout logs in Openshift.

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation

- [x] Output Encoding

- [x] Authentication and Password Management

- [x] Session Management

- [x] Access Control

- [x] Cryptographic Practices

- [ ] Error Handling and Logging

- [x] Data Protection

- [x] Communication Security

- [x] System Configuration

- [x] Database Security

- [x] File Management

- [x] Memory Management

- [x] General Coding Practices

### CODE REVIEW

File: pkg/logger/logger.go [hunk 1]
```diff
@@ -70,6 +70,9 @@ func (f *CustomCloudwatch) Format(entry *logrus.Entry) ([]byte, error) {
 		return nil, err
 	}
 
+	// Add newline to make stdout readable
+	j = append(j, 0x0a)
```
[the remark this answers is not in the document]
Comment (kylape) [reply pkg/logger/logger.go:74]: Yep, I tried `"\n"` first but it didn't work

File: pkg/config/config.go [hunk 2]
```diff
@@ -122,4 +125,23 @@ func Get() *IngressConfig {
...
+		cfg := clowder.LoadedConfig
+		broker := cfg.Kafka.Brokers[0]
+
+		if broker.Authtype != nil {
+			ingressCfg.KafkaUsername = *broker.Sasl.Username
+			ingressCfg.KafkaPassword = *broker.Sasl.Password
+			caPath, err := cfg.KafkaCa(broker)
+
+			if err != nil {
+				//TODO log some kind of error??
```
[the remark this answers is not in the document]
Comment (kylape) [reply pkg/config/config.go:139]: there's no logger.  maybe just panic?

File: Dockerfile [hunk 3]
```diff
@@ -1,19 +1,17 @@
...
-COPY . .
+FROM registry.redhat.io/ubi8/go-toolset as builder
 
 USER 0
+WORKDIR /go/src/app
+COPY go.mod go.sum .
+RUN go mod download
+COPY main.go .
+COPY pkg/ ./pkg/
+RUN CGO_ENABLED=1 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -a -o insights-ingress-go main.go
```
[the remark this answers is not in the document]
Comment (BlakeHolifield) [reply Dockerfile:9]: Ah, carry on as you were then.

File: pkg/config/config.go [hunk 4]
```diff
@@ -122,4 +126,23 @@ func Get() *IngressConfig {
 		UseSSL:               options.GetBool("UseSSL"),
 		UseClowder:           os.Getenv("CLOWDER_ENABLED") == "true",
 	}
+
+	if os.Getenv("CLOWDER_ENABLED") == "true" {
```
Comment (BlakeHolifield) [inline pkg/config/config.go:130]: Do we want to update the is to the `isClowderEnabled` method while this is open?

Comment (kylape) [reply pkg/config/config.go:130]: I think we can do this in a follow up PR imho

**Pull request merged**